File: scripts/load-env.js
#!/usr/bin/env node
/**
 * Script que carga variables del .env de la raíz del proyecto
 * Esto permite que frontend y backend lean del mismo .env centralizado
 * 
 * Uso: node scripts/load-env.js [comando]
 * Ejemplo: node scripts/load-env.js npm start
 */

const fs = require('fs');
const path = require('path');
const { spawn } = require('child_process');

// Ruta al .env en la raíz del proyecto
const envPath = path.join(__dirname, '..', '.env');

// Leer .env
if (!fs.existsSync(envPath)) {
    console.error(`❌ .env no encontrado en: ${envPath}`);
    process.exit(1);
}

const envContent = fs.readFileSync(envPath, 'utf8');
const lines = envContent.split('\n');

// Procesar cada línea y cargar en process.env
lines.forEach(line => {
    // Ignorar comentarios y líneas vacías
    if (line.trim().startsWith('#') || !line.trim()) return;
    
    // Parsear KEY=VALUE
    const [key, ...valueParts] = line.split('=');
    if (!key) return;
    
    const cleanKey = key.trim();
    const value = valueParts.join('=').trim();
    
    // Si es REACT_APP_* o no está en process.env, agregarlo
    if (cleanKey.startsWith('REACT_APP_') || !process.env[cleanKey]) {
        process.env[cleanKey] = value;
    }
});

console.log(`✓ Variables de .env cargadas desde: ${envPath}`);

// Si hay argumentos adicionales, ejecutar comando
const args = process.argv.slice(2);
if (args.length > 0) {
    const [command, ...cmdArgs] = args;
    const child = spawn(command, cmdArgs, {
        env: process.env,
        stdio: 'inherit',
        shell: true
    });
    
    child.on('exit', (code) => {
        process.exit(code);
    });
} else {
    console.log('Variables disponibles:');
    Object.keys(process.env)
        .filter(key => key.startsWith('REACT_APP_') || key.startsWith('DJANGO_') || key.startsWith('DATABASE_'))
        .forEach(key => {
            console.log(`  ${key}=${process.env[key]}`);
        });
}

File: scripts/load-env.py
#!/usr/bin/env python
"""
Script que cargaun variables del .env de la raíz del proyecto
Esto permite que frontend y backend lean del mismo .env centralizado

Uso: 
  python scripts/load-env.py manage.py runserver
  python scripts/load-env.py manage.py migrate
"""

import os
import sys
from pathlib import Path
from dotenv import load_dotenv

# Ruta al .env en la raíz del proyecto
PROJECT_ROOT = Path(__file__).parent.parent
ENV_FILE = PROJECT_ROOT / '.env'

if not ENV_FILE.exists():
    print(f"❌ .env no encontrado en: {ENV_FILE}")
    sys.exit(1)

# Cargar variables del .env centralizado
load_dotenv(ENV_FILE)

print(f"✓ Variables de .env cargadas desde: {ENV_FILE}")

# Si hay argumentos, ejecutar comando (ej: manage.py runserver)
if len(sys.argv) > 1:
    # Cambiar al directorio backend
    backend_dir = PROJECT_ROOT / 'backend'
    os.chdir(backend_dir)
    
    # Ejecutar comando
    cmd = sys.argv[1:]
    from django.core.management import execute_from_command_line
    try:
        execute_from_command_line(cmd)
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)
else:
    print('Variables disponibles:')
    for key in sorted(os.environ.keys()):
        if key.startswith(('DJANGO_', 'DATABASE_', 'REACT_APP_')):
            print(f"  {key}={os.environ[key]}")
